src/realm/util/string_buffer.hpp

```diff
@@ -134,14 +132,16 @@ inline const char* StringBuffer::data() const noexcept
 
 inline char* StringBuffer::c_str() noexcept
 {
+    static char zero = 0;
```

Comment:
I have to say I still don't like this particular solution to returning a pointer to an empty string. AFAIK every call to this function (and below) will now have some (thread-safe, even) check to see if `zero` has been initialised. @finnschiermer Do you have feedback here?